tests/topotests/multicast_pim_uplink_topo2/test_multicast_pim_uplink_topo2.py

```diff
@@ -0,0 +1,1350 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: ISC
+#
+# Copyright (c) 2023 by VMware, Inc. ("VMware")
+#
+
+"""
+Following tests are covered to test multicast pim sm:
+
+1. Verify changing RP address on DUT from Static to BSR , IIF and OIF
+    updated correctly
+2. Verify when mroute RPT and SPT path is difference
+3. Verify mroutes updated with correct OIL and IIF after shut / no shut of
+    upstream interface from DUT
+4. Verify mroutes updated with correct OIL and IIF after shut / no
+shut of downstream interface from FHR
+
+
+"""
+
+import os
+import sys
+import time
+from time import sleep
```

Comment:
Where is this used?

Reply:
Removed unused packages, updated code. Thank you.